src/dgram.rs

```diff
@@ -32,41 +32,42 @@ use crate::Result;
 /// Keeps track of Datagram frames.
 #[derive(Default)]
 pub struct DatagramQueue {
-    readable: VecDeque<Vec<u8>>,
-    writable: VecDeque<Vec<u8>>,
-    readable_len : usize,
-    writable_len : usize,
+    readable: DatagramVecQueue,
+    writable: DatagramVecQueue,
 }
 
-impl DatagramQueue {
-    pub fn new(readable_len : usize,
-        writable_len : usize,
-    ) -> Self {
-        DatagramQueue {
-            readable: VecDeque::new(),
-            writable: VecDeque::new(),
-            readable_len,
-            writable_len,
+#[derive(Default)]
+pub struct DatagramVecQueue {
```

Comment:
I hate the `DatagramVecQueue` name, but found none better so far, given that `DatagramQueue` is taken. Suggestions welcome.

Reply:
Simple question is, with this refactor do we need the current `pub struct DatagramQueue`? It seems to me if we go this path, we could continue simplifying by just removing that type and having a `conn.dgram_read_queue` and `conn.dgram_write_queue`.